### DISCUSSION TITLE
Changing `space` default increments from 4 to 16

### DISCUSSION THREAD

**Discussion by smblee** [chakra-ui/chakra-ui#3484] [Q&A] [answered]
I want to change the base `space` values from 4 to 16, i.e.

from:
```
space: {
    px: "1px",
    "0": "0",
    "1": "0.25rem",
    "2": "0.5rem",
    "3": "0.75rem",
    "4": "1rem",
    "5": "1.25rem",
    "6": "1.5rem",
    "8": "2rem",
    "10": "2.5rem",
    "12": "3rem",
    "16": "4rem",
    "20": "5rem",
    "24": "6rem",
    "32": "8rem",
    "40": "10rem",
    "48": "12rem",
    "56": "14rem",
    "64": "16rem",
  },
```

to

```
space: {
    px: "1px",
    "0": "0",
    "1": "0.0625rem",
    "2": "0.125rem",
    "3": "0.1875rem",
    "4": "0.25rem", 
   ...
    "16": "1rem",
   ...
  },
```

I want to do this because for all the theme configurations and component props i need to use space in, I can simply treat it as "px"-like value (e.g. 16 will translate to 16px like value in rem, which is 1rem). 

But I realized that if i change the default spaces, it messes up some of the built in component UIs, such as `Button`.

E.g.
![image](https://user-images.githubusercontent.com/9781750/109465905-c0125800-7a1d-11eb-9d96-2252a035e767.png)

How do people get around this issue? Will I have to fix all components that refer to `space` values? If so, what are all the components that use the global `space` values by default?

**Comment by with-heart** [comment, marked as answer]
Essentially you'd need to go through and adjust the `baseStyle` and `sizes` of most of the component themes listed in this file: https://github.com/chakra-ui/chakra-ui/blob/14156442c05edb9413aa8055be2da7b7a14a26e8/packages/theme/src/components/index.ts

Since `theme.space` is no longer a scale, you'd have to recreate that for each component.

**Reply by with-heart** [reply]
Are you sure that's what you want though? Scaled space values provide a consistent spacing grid, allowing designs to establish a consistent vertical rhythm, adequate whitespace, clear relationships between various sections of a page, etc. simply by utilizing relatively-sized values rather than needing to choose specific pixel values.

**Reply by smblee** [reply]
Thanks for the reply. As to am i sure if this is what i want, i am not sure if this is the "correct" solution for what I want to achieve (surely doesn't feel like it haha). The motivation stems from inconsistent/arbitrary values in "space" may be confusing during development (especially since we work off of Figma designs that come directly from design team).

I noticed all default "space" values (lineHeights, fontSizes, letterSpacings, sizes, border radius, etc.) use semantic names such as xl, s, 3, 4, etc. I guess for this to really work without the overhead of developers either constantly looking up semantic value mapping or trying different values until it "looks"right, we would need the Figma designs to either show the semantic values for spaces such as margins/widths/paddings. 

We want to make an app that uses em/rem and no px across (not sure if this is practical or necessary, would love to hear your thoughts on this, but we believe it's good for accessibility for scaling based on root font size). Since currently we work with designs that give us pixel values, I wanted to use `space` values to avoid introducing friction for developers for reasons stated above, and also to avoid converting px to rem (1:16) every time we need to hard code them.

Again, it doesn't seem like the "suggested" approach and would love to hear your thoughts on your suggestions. We would love Chakra to work as a base library for our apps at the scale we are in.

**Reply by with-heart** [reply]
The way we accomplished this at work was to involve our UX/design team in the conversation around Chakra from the start to make sure that they were bought in and understood that the component library would be driven by Chakra under the hood. This means that, while their designs use pixel values (since that's just how Sketch, Figma, etc. work) for various things, they make sure that the size/space pixel values come from the Chakra scale, so that developers just need to know the "divide by 4" rule for determining which size/scale value to use.

Essentially, it's about establishing a shared language between developers and designers, and making sure that that language matches what's in your Chakra theme, whether that's the default scales that come with Chakra or something custom.

**Reply by smblee** [reply]
Fair point. I think there needs to be an agreement with engineering and design. I agree spaces can be represented in less granular increments as long as they are agreed upon. Thanks for the insight and i'll close this thread!
